### PR TITLE
market history chart load async

### PR DESCRIPTION
**Pull request by GuSilverFlame**
part of #1161 

### Enhancements
* load market history chart asynchronously

### screenshots
image

image

### CODE REVIEW

File: apps/block_scout_web/assets/css/components/_dashboard-banner.scss [hunk 1]
```diff
@@ -63,6 +63,8 @@
 .dashboard-banner-chart {
   grid-area: chart;
   padding: 1rem 1rem 1rem 0;
+  width: 350px;
```
Comment (amandasposito) [inline apps/block_scout_web/assets/css/components/_dashboard-banner.scss]: what happens with this graph in the mobile?

File: apps/block_scout_web/test/block_scout_web/controllers/chain/market_history_chart_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  describe "GET show/2" do
+    test "returns error when not an ajax request" do
+      path = market_history_chart_path(BlockScoutWeb.Endpoint, :show)
+
+      conn = get(build_conn(), path)
+
+      assert conn.status == 422
+    end
+
+    test "return ok, when request is ajax" do
```
Comment (w-sanches) [inline apps/block_scout_web/test/block_scout_web/controllers/chain/market_history_chart_controller_test.exs]:
```suggestion
    test "returns ok when request is ajax" do
```

File: apps/block_scout_web/assets/css/components/_dashboard-banner.scss (hunk 1 above)
Comment (GuSilverFlame) [reply apps/block_scout_web/assets/css/components/_dashboard-banner.scss]: it is hidden on mobile

File: apps/block_scout_web/lib/block_scout_web/controllers/chain/market_history_chart_controller.ex [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      exchange_rate = Market.get_exchange_rate(Explorer.coin()) || Token.null()
+
+      market_history_data =
+        30
+        |> Market.fetch_recent_history()
+        |> case do
+          [today | the_rest] -> [%{today | closing_price: exchange_rate.usd_value} | the_rest]
+          data -> data
+        end
+        |> encode_market_history_data
```
Comment (w-sanches) [inline apps/block_scout_web/lib/block_scout_web/controllers/chain/market_history_chart_controller.ex]:
```suggestion
        |> encode_market_history_data()
```

Just to keep the pattern some lines above.

**Comment by acravenho** [comment]
screen shot 2018-12-27 at 3 52 42 pm

Dai Chain using the `SUPPLY_MODULE` and `SOURCE_MODULE`. The market cap is showing but when hovering over the chart, it's not working.

**Comment by acravenho** [comment]
Normal chain market cap not working. The price is displaying when hovering over the chart but not on the section below the chart

screen shot 2018-12-27 at 3 58 08 pm

**Comment by GuSilverFlame** [comment]
> Normal chain market cap not working. The price is displaying when hovering over the chart but not on the section below the chart
> 
> screen shot 2018-12-27 at 3 58 08 pm

this one's weird, I tried the request to https://api.coinmarketcap.com/v1/ticker/ and POA doesn't seem to be coming in the response anymore, that messes up the chart, the exchange price and the market cap because when we find nothing we replace the price with a zero and I tested with ethereum(mainnet) since I saw that it still comes from the endpoin, the chart works ok for that one
I really don't know why POA is not coming anymore from the endpoint

as for dai, it seems to be working fine here.
are you testing on a database with info from the dai chain or just with the coin name and configs for dai on a sokol database? 
I had the same problem when I tried to test the dai chain chart but wth a sokol DB because the mint and burn operations for dai don't happen in sokol(the burn ones do but the char goes completely crazy when it finds negative values and just says "less than $0,000000000000001"(probably without so many zeroes but something like that)

**Comment by acravenho** [comment]
Nice work @GuSilverFlame

**Pull request merged**